tag_filter.py

```diff
@@ -0,0 +1,182 @@
+
+    # returns data with device uuid and device name
+    def tag_name(self): 
+            endpoint = self.api_url + '/api/proximity/getMinimalProximityStateList'
+            payload = {
+            }
+            resp = self.api_sess.post(endpoint, json=payload,
+                            verify=False)
+            order_content = resp.content.decode('utf8')
+            tag_name_data = json.loads(order_content)
+            return tag_name_data
+
+    # returns data with address
+    def locations(self):
+        endpoint = self.api_url + "/api/location/getLocations"
+        payload = {
+        }
+        resp = self.api_sess.post(endpoint, json=payload,
+                       verify=False)
+        order_content = resp.content.decode('utf8')
+        # Load the JSON to a Python list & dump it back out as formatted JSON
+        location_data = json.loads(order_content)
+        return location_data
+
+    # converts lat and lon to street address
+    def location_data_print(self, location_data, lat_1):
+        for value in location_data['locations']:
+            lat_2 = round(value['latitude'], 2)
+            if lat_1 == lat_2:
+                address = value['address1'] + ' ' + value['address2']
+                return address
+
+    def tag_data(self):    # returns data that should be put in CSV file
+        endpoint = self.api_url + "/api/proximity/getLocomotionEventsForTag"
+        self.tag_name_data = self.tag_name()
+        self.uuid = self.tag_name_convert()
+
+        if self.args.startTime:
+            self.start_time = self.milliseconds_time(self.args.startTime)
+        else:
+            self.start_time = int(round((time.time() - (3600 * 24)) * 1000))  # start time defaults to 24 hours ago
+        if self.args.endTime:
+            self.end_time = self.milliseconds_time(self.args.endTime)
+        else:
+            self.end_time = int(round(time.time() * 1000))    # end time defaults to now
+
+        payload = {
+            "movementFilter": self.args.movement,
+            "tagUuid": self.uuid,
+            "createdBeforeMs": self.end_time,
+            "createdAfterMs": self.start_time
+        }
+        resp = self.api_sess.post(endpoint, json=payload,
+                        verify=False)
+        order_content = resp.content.decode('utf8')
+        data = json.loads(order_content)
+        return data
+
+    def csv_add(self, value):
+        self.csv_data.append([])
+        self.real_name = self.tag_uuid_convert(value['tagUuid'], self.tag_name_data)
+        self.csv_data[self.count].append(self.real_name)
+        lat_1 = round(value['gpsLocation']['lat'], 2)
+        address = self.location_data_print(self.location_data, lat_1)
+        self.csv_data[self.count].append(address)
+        self.csv_data[self.count].append(value['movement'])
+        self.csv_data[self.count].append(self.human_time(value['timestampMs']))
+        self.csv_data[self.count].append(self.count + 1)
```

Comment:
Unsure what this column does - it is not tied to anything else and is unnecessary?